dspace-api/src/main/java/org/dspace/content/dao/impl/ItemDAOImpl.java

```diff
@@ -103,6 +118,111 @@
         return iterate(query);
     }
 
+    enum OP {equals,not_equals,like,not_like,contains,doesnt_contain,exists,doesnt_exist,matches,doesnt_match;}
+    
+    @Override
+    public Iterator<Item> findByMetadataQuery(Context context, List<String> query_field, List<String> query_op, List<String> query_val, List<String> collection_ids, String regexClause) throws SQLException {
```

Comment:
There is to much "business" logic in this DAO method, the DAO should just be about running a database query. The transformation of strings to UUIDs, splitting of metadata value should go to the service layer.
The reason for this being if anybody else wants a different DAO then hibernate they would have to reimplement all this logic as well.